boards/arm/stm32h7/nucleo-h743zi/scripts/flash.ld

```diff
@@ -172,9 +172,12 @@ SECTIONS
 
     /* Emit the the D3 power domain section for locating BDMA data */
 
-    .sram4 :
-    {
-    } > sram4
+	.sram4_reserve :
+	{
+		_sram4_reserve_start = ABSOLUTE(.);
```

Comment:
I am wondering if we could use a name here that says what it is.
```
_sram4_heapexclude_start = ABSOLUTE(.);
*(.sram4)
_sram4_heapexclude_end = ABSOLUTE(.);
_sram4_heap_start  = ABSOLUTE(.);
```

Reply:
Agreed. Better naming makes it far more clear what this is for.

I think it also needs a `. = ALIGN(4)` in case of oddly sized data being placed here.

Right now I have it as this:

```
	.sram4_reserve (NOLOAD) :
	{
		_sram4_heapexclude_start = ABSOLUTE(.);
		*(.sram4)
		. = ALIGN(4);
		_sram4_heapexclude_end = ABSOLUTE(.);
		_sram4_heap_start = ABSOLUTE(.);
	} > sram4
```

But I think it can be improved/simplified further:

_sram4_heapexclude_start really serves no purpose other than documentation.

Also, because _sram4_heapexclude_end and _sram4_heap_start are the same address, we could remove _sram4_heapexclude_end and keep just _sram4_heap_start.

This will simplify the code changes in stm32_allocateheap.c as well (by removing a bunch of unnecessary symbols).

I'll make this change... let me know what you think. I can always change it back if we don't like it.